src/DotNetBridge/NativeDataView.cs

```diff
@@ -1015,30 +943,20 @@ public override void CopyOut(long index, Batch batch, ref uint value)
                     _getter(Data, ColIndex, index, out value);
                 }
 
-                public override IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes()
-                {
-                    var res = base.GetMetadataTypes();
-                    if (_keyValuesType != null)
-                        res = res.Prepend(_keyValuesType.GetPair(MetadataUtils.Kinds.KeyValues));
-                    return res;
-                }
-
-                public override ColumnType GetMetadataTypeOrNull(string kind)
+                public override Schema.Metadata Metadata
                 {
-                    Contracts.AssertNonEmpty(kind);
-                    if (kind == MetadataUtils.Kinds.KeyValues && _keyValuesType != null)
-                        return _keyValuesType;
-                    return base.GetMetadataTypeOrNull(kind);
-                }
+                    get
+                    {
+                        if (_keyValuesType != null)
+                        {
+                            var kvPair = _keyValuesType.GetPair(MetadataUtils.Kinds.KeyValues);
+                            var metadataBuilder = new MetadataBuilder();
+                            metadataBuilder.Add(kvPair.Key, kvPair.Value, _getKeyValues);
```

Comment:
>Add [](start = 44, length = 3)

MetadataBuilder has a method AddKeyValues.

Reply:
Using AddKeyValues instead.

---
In reply to: [250274845](https://github.com/Microsoft/NimbusML/pull/90#discussion_r250274845) [](ancestors = 250274845)